comp/core/sysprobeconfig/config.go

```diff
@@ -0,0 +1,151 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package sysprobeconfig
+
+import (
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/internal"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// cfg implements the Component.
+type cfg struct {
+	// this component is currently implementing a thin wrapper around pkg/config,
+	// and uses globals in that package.
+
+	// warnings are the warnings generated during setup
+	warnings *config.Warnings
+}
+
+type dependencies struct {
+	fx.In
+
+	Params internal.BundleParams
+}
+
+func newConfig(deps dependencies) (Component, error) {
+	warnings, err := setupConfig(
+		deps.Params.SysProbeConfFilePath,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cfg{warnings}, nil
+}
+
+func (c *cfg) IsSet(key string) bool {
```

Comment:
The code is similar to comp/core/config/config.go. A common implementation should be used for both.